user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleDetailsComponent.java

```diff
@@ -155,24 +160,32 @@ private static Grid<SamplePreview> createSampleGrid() {
   }
 
   public void onSearchFieldValueChanged(String searchValue) {
-    updateSampleGridDataProvider(context.experimentId().orElseThrow(), searchValue);
+    updateSampleGridDataProvider(context.projectId().orElseThrow(), context.experimentId().orElseThrow(), searchValue);
   }
 
-  private void updateSampleGridDataProvider(ExperimentId experimentId, String filter) {
+  private void updateSampleGridDataProvider(ProjectId projectId, ExperimentId experimentId,
+      String filter) {
     sampleGrid.setItems(query -> {
       List<SortOrder> sortOrders = query.getSortOrders().stream().map(
               it -> new SortOrder(it.getSorted(), it.getDirection().equals(SortDirection.ASCENDING)))
           .collect(Collectors.toList());
       // if no order is provided by the grid order by last modified (least priority)
       sortOrders.add(SortOrder.of("sampleCode").ascending());
-      return sampleInformationService.queryPreview(experimentId, query.getOffset(),
-          query.getLimit(), List.copyOf(sortOrders), filter).stream();
+      return Objects.requireNonNull(
+          asyncProjectService.getSamplePreviews(projectId.value(), experimentId.value(),
+                  query.getOffset(), query.getLimit(), List.copyOf(sortOrders), filter).collectList()
+              .doOnError(RequestFailedException.class, this::handleRequestFailed)
+              .block()).stream();
     });
     sampleGrid.getLazyDataView().addItemCountChangeListener(
         countChangeEvent -> setSampleCount((int) sampleGrid.getLazyDataView().getItems().count()));
     sampleGrid.recalculateColumnWidths();
   }
 
+  private void handleRequestFailed(RequestFailedException e) {
+
```

Comment:
What was done before when the failure case occurred?

Reply:
nothing? :D